dhis-2/dhis-services/dhis-service-program-rule/src/main/java/org/hisp/dhis/programrule/api/RuleEngineEffects.java

```diff
@@ -147,14 +150,23 @@ private static List<ValidationEffect> mapValidationEffect(List<RuleEffect> effec
 
   private static List<NotificationEffect> mapNotificationEffect(List<RuleEffect> effects) {
     return effects.stream()
-        .filter(e -> NotificationAction.contains(e.getRuleAction().getType()))
+        .filter(
+            e ->
+                e.getRuleAction().getType().equals(SENDMESSAGE.name())
```

Comment:
Is the type ever null? Putting constants first is usually safer so I think we should do it regardless of the answer 😁